vdslib/src/vespa/vdslib/distribution/distribution.cpp

```diff
@@ -459,12 +460,12 @@ Distribution::getIdealNodes(const NodeType& nodeType,
                 auto it = tmpResults.begin();
                 for (; it != tmpResults.end(); ++it) {
                     if (score > it->_score) {
-                        tmpResults.insert(it, ScoredNode(nodes[j], nodeState.getReliability(), score));
+                        tmpResults.insert(it, ScoredNode(score, nodes[j], nodeState.getReliability()));
                         break;
                     }
                 }
                 if (it == tmpResults.end()) {
-                    tmpResults.emplace_back(nodes[j], nodeState.getReliability(), score);
+                    tmpResults.emplace_back(score, nodes[j], nodeState.getReliability());
```

Comment:
Hmm is it technically possible for `it` to be invalidated by the `insert` above, i.e. comparing a bogus iterator with `end`? It _probably_ won't be since it's pre-reserved, but I'm not sure what guarantees are given here... Maybe check on something else than the iterator to be sure? If this is pulled out into its own function then could return instead of break and avoid the check entirely.

Reply:
Fixed